alchemical_analysis/alchemical_analysis.py

```diff
@@ -170,28 +171,63 @@ def uncorrelate(sta, fin, do_dhdl=False):
-      g[k] = pymbar.timeseries.statisticalInefficiency(dhdl_sum)
-      indices = sta[k] + numpy.array(pymbar.timeseries.subsampleCorrelatedData(dhdl_sum, g=g[k])) # indices of uncorrelated samples
-      N = len(indices) # number of uncorrelated samples
-      # Handle case where we end up with too few.
-      if N < P.uncorr_threshold:
+
+   UNCORR_OBSERVABLE = {'Gromacs':P.uncorr, 'Amber':'dhdl', 'Sire':'dhdl'}[P.software.title()]
+
+   if UNCORR_OBSERVABLE == 'dhdl':
+
+      for k in range(K):
+         # Sum up over the energy components; notice, that only the relevant data is being used in the third dimension.
+         dhdl_sum = numpy.sum(dhdlt[k,:,sta[k]:fin[k]], axis=0)
+         # Determine indices of uncorrelated samples from potential autocorrelation analysis at state k
+         # (alternatively, could use the energy differences -- here, we will use total dhdl).
+         g[k] = pymbar.timeseries.statisticalInefficiency(dhdl_sum)
+         indices = sta[k] + numpy.array(pymbar.timeseries.subsampleCorrelatedData(dhdl_sum, g=g[k])) # indices of uncorrelated samples
+         N = len(indices) # number of uncorrelated samples
+         # Handle case where we end up with too few.
+         if N < P.uncorr_threshold:
+            if do_dhdl:
+               print "WARNING: Only %s uncorrelated samples found at lambda number %s; proceeding with analysis using correlated samples..." % (N, k)
+            indices = sta[k] + numpy.arange(len(dhdl_sum))
+            N = len(indices)
+         N_k[k] = N # Store the number of uncorrelated samples from state k.
+         if not (u_klt is None):
+            for l in range(K):
+               u_kln[k,l,0:N] = u_klt[k,l,indices]
          if do_dhdl:
-            print "WARNING: Only %s uncorrelated samples found at lambda number %s; proceeding with analysis using correlated samples..." % (N, k)
-         indices = sta[k] + numpy.arange(len(dhdl_sum))
-         N = len(indices)
-      N_k[k] = N # Store the number of uncorrelated samples from state k.
-      if not (u_klt is None):
-         for l in range(K):
-            u_kln[k,l,0:N] = u_klt[k,l,indices]
-      if do_dhdl:
-         print "%6s %12s %12s %12.2f" % (k, fin[k], N_k[k], g[k])
-         for n in range(n_components):
-            dhdl[k,n,0:N] = dhdlt[k,n,indices]
+            print "%6s %12s %12s %12.2f" % (k, fin[k], N_k[k], g[k])
+            for n in range(n_components):
+               dhdl[k,n,0:N] = dhdlt[k,n,indices]
+
+   if UNCORR_OBSERVABLE == 'dE':
+
+      for k in range(K):
+         ## Sum up over the energy components; notice, that only the relevant data is being used in the third dimension.
+         #dhdl_sum = numpy.sum(dhdlt[k,:,sta[k]:fin[k]], axis=0)
+         # Determine indices of uncorrelated samples from potential autocorrelation analysis at state k
+         # (alternatively, could use the energy differences -- here, we will use total dhdl).
```

Comment:
Thanks for taking care of this!  You probably want to clean up this comment so it's clearer -- right now it's a repeat from above.